src/odemis/driver/andorshrk.py

```diff
@@ -631,6 +630,9 @@ def Initialize(self):
         else:
             path = ""
 
+        if not isinstance(path, bytes):  # Python 3
+            path = os.fsencode(path)
```

Comment:
Same comment here: `fsencode` doesn't exist in Python 2, are you sure `not isinstance(path, bytes)` will always return `False` for Python 2?